src/server/scriptInfo.ts

```diff
@@ -294,6 +294,11 @@ namespace ts.server {
          * All projects that include this file
          */
         readonly containingProjects: Project[] = [];
+        /**
+         * @internal
+         * Projects whose auto-import providers include this file
+         */
+        private auxiliaryFileContainingProjects: Project[] | undefined;

```

Comment:
ScriptInfo needs to track what projects’ auto-import provider programs contain it so they can be updated when edited manually. We can’t track this in `containingProjects`, because it would mess up the calculation of a script’s “default project.” (When a file is in the auto-import program but not the main program, and you open that file in your editor for some reason, it should be part of an InferredProject, not the ConfiguredProject that might eventually auto-import it.) Thanks @sheetalkamat for advising me to look into this scenario.